### ISSUE TITLE
Dialogues glitch on black screen

### ISSUE THREAD

**Issue by nagman** [chrismaltby/gb-studio#879]
**Describe the bug**
When on a black screen (doesn't matter if it's a top town 2D or a Logo), my dialogues appears twice with sort of a glitch.
And there's no UI for the dialogues box.

**To Reproduce**
I haven't managed to reproduce it on a new project.
But on my current project, which I've migrated from v2, every new scene with a black background shows this bug.
Even if I create a single scene with only dialogues, and no link to the other scenes, it shows the glitch.

**Screenshots**
![image](https://user-images.githubusercontent.com/16116486/146273877-d023397a-42b5-4e62-bb6a-5e795577661d.png)

**Platform (please complete the following information):**
 - OS: macOS 11.6
 - App version: 3.0.0

**Additional context**
I wonder if it's not related to the migration from v2 to v3.
I hope it's not 'cause I don't want to recreate my whole project once again.
Except if there's a way to customise the gravity settings with the v2, because my project worked very well before the migration - I just needed to configure the platform engine and I did the upgrade for this feature.

**Comment by maxoakland** [comment]
You can configure the physics settings in 2.0 beta 5. It was the latest version before 3.0 was released

**Comment by zenzoa** [comment]
I might have something related to this. I'm working on a new GB Studio 3.0.1 project, currently don't have a whole lot in it. I have one top-down scene with one actor and the player, and some on-interact dialog events on the actor. This is what happens when I trigger the dialog:
![image](https://user-images.githubusercontent.com/10480278/146680788-c0c31487-3cb1-457d-9014-dd3b8140a1f7.png)

**Comment by pau-tomas** [comment]
@nagman @zenzoa will any of you mind sharing your project here? That way it'll be easier to debug what might be happening. Thanks!

**Comment by zenzoa** [comment]
Sure! Here you go:
[Halo Finite.zip](https://github.com/chrismaltby/gb-studio/files/7741124/Halo.Finite.zip)

**Comment by nagman** [comment]
Sure! Here it is:
[Archive.zip](https://github.com/chrismaltby/gb-studio/files/7741125/Archive.zip)

**Comment by untoxa** [comment]
you can not use dialogs with the logo scene, because tiles for rendering of text are used for background.

**Comment by nagman** [comment]
> you can not use dialogs with the logo scene, because tiles for rendering of text are used for background.

The bug happens wether it's a Logo scene or a top down 2D.

**Comment by pau-tomas** [comment]
I did some testing with the projects and what it seems to be happening is that the same background is used for both log and non-logo scenes and that might be making the non-logo scenes to act as logo ones? It's harder to reproduce on nagman's project but in zenzoa's changing the background of the Intro-3 to the one in Intro-1 (a non-logo scene) fixes the issue.

**Comment by zenzoa** [comment]
Thanks for taking a look! Thankfully I'd planned on changing the background in the logo scene anyway, and since I gave it its own bg, the issue hasn't cropped up again.

**Comment by untoxa** [comment]
@pau-tomas yes. background for the logo scene uses special pre-defined map that uses 360 tiles, including those, which are reserved for UI. if you set the very same background for the normal scene (i mean in the source, not in the IDE), that will result into this. so backgrounds for logo must not be mixed with backgrounds for other scenes, even if there are little used tiles. or if there are less or equal than 192, then regular map should be emitted and then it is kind of "convertible". regular map may be used for the logo scene, but not the opposite.

**Comment by nagman** [comment]
@untoxa I'm not sure to understand.

So that means that a background asset for black screens (I mean the source file, the `black.png`) should only be used in either Logo scenes or Top down 2D scenes?

I tried to change all my black scenes into Logo scenes but it changed nothing.

**But changing them to Top down 2D solved the issue!**

And I've not understood the second part:
> or if there are less or equal than 192, then regular map should be emitted and then it is kind of "convertible". regular map may be used for the logo scene, but not the opposite.

**Comment by pau-tomas** [comment]
Text is expected to glitch in logo scenes no matter what background they use. However if one scene that is not logo uses the background png of one that is logo it'll also glitch as the background tileset is compiled expecting it'll be used only on logo scenes.

To put it another way if you want a logo scene with a black background and a non-logo scene also with non-black background you should have two pngs, even if they look the same and use them on each scene.

That said, it doesn't make much sense to use logo for a black background scene as the point of logo is to be able to use as many tiles as possible :)

**Comment by untoxa** [comment]
currently, logo background, even empty, consist of 360 tiles (5760 bytes) plus 360 bytes of tile map. so logo scene should be used only where you actually need it.

**Comment by nagman** [comment]
@pau-tomas thanks! It's crystal clear.
In brief:
- Logo scenes
